geode-core/src/main/java/org/apache/geode/internal/cache/execute/FunctionContextImpl.java

```diff
@@ -37,20 +38,25 @@
 
   private String functionId = null;
 
+  private Cache cache = null;
+
   private ResultSender resultSender = null;
 
   private final boolean isPossDup;
 
   public FunctionContextImpl(final String functionId, final Object args,
       ResultSender resultSender) {
-    this.functionId = functionId;
-    this.args = args;
-    this.resultSender = resultSender;
-    this.isPossDup = false;
+    this(null, functionId, args, resultSender, false);
+  }
+
+  public FunctionContextImpl(final Cache cache, final String functionId, final Object args,
+      ResultSender resultSender) {
+    this(cache, functionId, args, resultSender, false);
   }
 
-  public FunctionContextImpl(final String functionId, final Object args, ResultSender resultSender,
-      boolean isPossibleDuplicate) {
+  public FunctionContextImpl(final Cache cache, final String functionId, final Object args,
+      ResultSender resultSender, boolean isPossibleDuplicate) {
```

Comment:
Looking at the places where it's grabbing the cache to create the FunctionContext, it's all using GemfireCacheImpl.getInstance(). I am wondering we should just have the implementation of getCache() in FunctionContextImpl to be simply GemfireCacheImpl.getInstance(), then we don't need to change all the constructor signatures and keep a member variable of the cache in the context.